pkg/services/network/network_test.go

```diff
@@ -470,7 +500,124 @@ var _ = Describe("Network provider", func() {
 				err = np.VerifyNetworkStatus(ctx, clusterCtx, vnetObj)
 
 				expectedErrorMessage := fmt.Sprintf("virtual network ready status is: '%s' in cluster %s. reason: %s, message: %s",
-					"False", apitypes.NamespacedName{Namespace: dummyNs, Name: dummyCluster}, testVnetNotRealizedReason, testVnetNotRealizedMessage)
+					"False", apitypes.NamespacedName{Namespace: dummyNs, Name: dummyCluster}, testNetworkNotRealizedReason, testNetworkNotRealizedMessage)
+				Expect(err).To(MatchError(expectedErrorMessage))
+				Expect(conditions.IsFalse(clusterCtx.VSphereCluster, vmwarev1.ClusterNetworkReadyCondition)).To(BeTrue())
+			})
+		})
+
+		Context("with nsx-vpc network provider and subnetset exists", func() {
+			BeforeEach(func() {
+				client = fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(configmapObj, systemNamespaceObj).Build()
+				vpcNp, _ = NSXTVpcNetworkProvider(client).(*nsxtVPCNetworkProvider)
+				np = vpcNp
+				err = np.ProvisionClusterNetwork(ctx, clusterCtx)
+			})
+
+			It("should not update subnetset", func() {
+				var initialSubnetSetSpec nsxopv1.SubnetSetSpec
+
+				// Fetch the SubnetSet before the operation
+				initialSubnetSet := &nsxopv1.SubnetSet{}
+				err = client.Get(ctx, apitypes.NamespacedName{
+					Name:      dummyCluster,
+					Namespace: dummyNs,
+				}, initialSubnetSet)
+				Expect(err).NotTo(HaveOccurred())
+				initialSubnetSetSpec = initialSubnetSet.Spec
+
+				// Presumably there's code here that might modify the SubnetSet...
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
```suggestion
```
err was checked above already

Reply:
I think this comment got lost. We still have the duplicate err check

Reply:
Dropped the dup check now.